persistence-api/src/main/java/io/stargate/db/DbActivator.java

```diff
@@ -0,0 +1,71 @@
+package io.stargate.db;
+
+import com.codahale.metrics.MetricRegistry;
+import io.stargate.db.cdc.CDCProducer;
+import io.stargate.db.cdc.CDCService;
+import io.stargate.db.cdc.CDCServiceImpl;
+import org.osgi.framework.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class DbActivator implements BundleActivator, ServiceListener {
+  private static final Logger logger = LoggerFactory.getLogger(DbActivator.class);
+  public CDCService cdcService;
+  private boolean started;
+  private BundleContext context;
+
+  @Override
+  public void start(BundleContext context) throws InvalidSyntaxException {
```

Comment:
I would add synchronized to all methods: those will be called very rarely so overhead would be low.

Reply:
Done!